### DISCUSSION TITLE
The unified file structure way in `blitz generate` cli?

### DISCUSSION THREAD

**Discussion by esmadrider** [blitz-js/blitz#2367] [RFCs]
Hi, guys.

When I started to learn Blitz.js, at the ***[Tutorial chapter](https://blitzjs.com/docs/tutorial)*** , `blitz generate` CLI generate two models for me: `app/questions` and `app/pages`, so the file structure goes to:

```
.
|-- api
|-- auth
|-- choices
|-- core
|-- pages
|   |-- questions
|-- questions
|-- users
```

It is a little hard to find model code. Is there a way to classify models into one `models` folder like this:?

```
.
|-- api
|-- core
|-- models
|   |-- auth
|   |-- choices
|   |-- questions
|   |-- users
|-- pages
|   |-- questions
```

I am very glad to support the second way :)

**Comment by flybayer** [comment]
After the code is generated, you can move the files around however you like.

For generating the code, you can do this: `blitz generate all models/projects`

**Reply by esmadrider** [reply]
> After the code is generated, you can move the files around however you like.
> 
> For generating the code, you can do this: `blitz generate all models/projects`

Ohh, thx! I forget we can directly use `blitz generate all ${scope}/projects`, it's more convenient now.

**Comment by esmadrider** [comment]
But I got another question about how to combine two folders `app/pages/projects` and `app/modles/projects` together into one folder`app/modles/projects` using `blitz generate` CLI.

----

In the ***[File Structure chapter][1]*** said an important opinion - live together:

> Files that change together should live together.

> Split While you can place any route files here, we recommend placing route files inside the `app/`. But if you want, you can instead place all your route files in this top level folders instead of being spread out in various folders inside `app`.

But `blitz generate all models/project` generate two `projects/` folder, one in `app/models/projects/`, one in `app/pages/projects`.

I am confused, according to the guiding principles should not it all generate into `app/models/projects/`? Now I have to maintain two code places for one `projects` model...

----

Thus, what is the good practice way should I follow? `blitz generate all` generate two folders which are split, but blitz also recommends relative code all in one...

How to use `blitz generate all projects` to generate the following file structure automatically and directly?

```
app/projects/pages/[projectId]/edit.tsx
app/projects/pages/[projectId].tsx
app/projects/pages/index.tsx
app/projects/pages/new.tsx
app/projects/components/ProjectForm.tsx
app/projects/queries/getProject.ts
app/projects/queries/getProjects.ts
app/projects/mutations/createProject.ts
app/projects/mutations/deleteProject.ts
app/projects/mutations/updateProject.ts
```

Looking forward to your reply 😄

[1]: https://blitzjs.com/docs/file-structure

**Reply by flybayer** [reply]
We originally did that but switched to a unified pages folder as it's more natural for most people.

For example, the paths you posted are incorrect, it would actually be this:

```
app/projects/pages/projects/[projectId]/edit.tsx
app/projects/pages/projects/[projectId].tsx
app/projects/pages/projects/index.tsx
app/projects/pages/projects/new.tsx
```

But we can add a config option that enables that behavior for generators.
